lib/ClangImporter/ClangImporter.cpp

```diff
@@ -960,8 +960,7 @@ bool ClangImporter::canReadPCH(StringRef PCHFilename) {
   // when there is no need for update.
   clang::CompilerInstance CI(Impl.Instance->getPCHContainerOperations(),
                              &Impl.Instance->getModuleCache());
-  auto invocation =
-      std::make_shared<clang::CompilerInvocation>(*Impl.Invocation);
+  auto invocation = Impl.Invocation;
```

Comment:
Actually, I wonder if this change makes this function really error prone to use. Previously, we did not modify `*Impl.Invocation`, but after this change we do all sorts of mutations that are persisted after this method call. Based on the name of this function as a caller I would definitely not anticipate these effects.

Reply:
I didn't realize all the shared_ptr inside CompilerInvocationBase actually has a deep copy constructor. Hmm, I need to find a different way to preserve CAS.